install.sh

```diff
@@ -76,7 +76,8 @@ if [ $upgrade -eq 1 -a $makerfaire2018 -eq 0 -a -d ${home_dir}/wm8960 ]; then
   echo "Updating sound driver - 2/14" > /tmp/pynab.upgrade
   cd ${home_dir}/wm8960
   sudo chown -R ${owner} .git
-  pull=`git pull`
+  # Making sure we keep the up-to-date status message consistent no matter the rebase default behavior
+  pull=$(LANGUAGE=en git pull --no-rebase)
```

Comment:
Likewise, I don't understand why we would like LANGUAGE=en here.
This code is not meant to be executed from an interactive session.

Reply:
See better implementaitons such as:
https://unix.stackexchange.com/questions/155046/determine-if-git-working-directory-is-clean-from-a-script/394674#394674

Reply:
Good point that there may be some porcelain/plumbing trick we can use to have a consistent status no matter what the locale & git config is active in the OS.
However, the git diff-index discussed in your link seems to work only where the local clone is ahead (or with untracked files) compared to remote. When I tried to test it with a local clone behind the remote (by doing a reset --hard to a previous commit & confirming that status reports a branch behind by X commits) the diff-index seems to return 0 the same way that when the branch is up-to-date.

I will keep on searching in the reference documentation until I hit something that could provide us a reliable way to tell if we have fresh code we want to rebuild ;)

Reply:
So I ended up decomposing the action in 3: doing a git fetch before, checking the local/remote discrepancy with status --porcelain (safest) and apply the pull if there is something to get and rebuild.

When there is no remote/local difference, the output looks like:
`ᐅ git status --branch --porcelain 
## bulleye_upgrade_fixes...origin/bulleye_upgrade_fixes`

And where there is a difference:
`ᐅ git status --branch --porcelain
## bulleye_upgrade_fixes...origin/bulleye_upgrade_fixes [behind 1]`

As it is a change of strategy in the code, I kept the change in a separate commit for now until it is considered satisfactory (and then I can rebase everything into a single commit)

Reply:
> Likewise, I don't understand why we would like LANGUAGE=en here. This code is not meant to be executed from an interactive session.

Just my 2 cts. If you want to have consistent messages in your scripts tests, use LC_ALL=C, not LANGUAGE=en. LANGUAGE=en will still depend on the English flavors (en-GB, en-US, en-CA...)

See https://unix.stackexchange.com/questions/87745/what-does-lc-all-c-do for a good explanation.